ddprof-exporter/src/containerid.rs

```diff
@@ -0,0 +1,125 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2021-Present Datadog, Inc.
+
+use regex::Regex;
+use std::error;
+use std::fmt;
+use std::fs::File;
+use std::io::{BufRead, BufReader};
+use std::path::Path;
+
+// Extract container id from /proc/self/group
+// Source: https://github.com/DataDog/dd-trace-go/blob/v1/internal/container.go
+
+const CGROUP_PATH: &str = "/proc/self/cgroup";
+
+const UUID_SOURCE: &str =
```

Comment:
I am also very confused by these magical regexes. I found some documentation:
https://github.com/DataDog/architecture/blob/master/rfcs/apm/agent/containers-tagging/rfc.md
https://docs.google.com/document/d/1nxc4nN2NWy0w0Sez4FG3jmqfw_FT_VtMroHfqYp6qio/edit#
Each language seems to implement *approximately* the same process, but still not quite exactly the same.
PHP has some king of precedence of task ID over container ID (https://github.com/DataDog/dd-trace-php/blob/master/components/container_id/container_id.c#L228) because for FarGate we return a Task ID instead of a Container ID. But this precedence is not implemented in other languages and I am not sure it's possible to both have container IDs and task IDs in the same `/proc/self/cgroup`. @morrisonlevi WDYT ? 
* Java has not such precedence, but returns the result of the last matching line: https://github.com/DataDog/dd-trace-java/blob/master/utils/container-utils/src/main/java/datadog/common/container/ContainerInfo.java#L164
* Python returns the result of the first matching line: https://github.com/Yun-Kim/dd-trace-py/blob/1.0-dev/ddtrace/internal/runtime/container.py#L105. Go does the same: https://github.com/DataDog/dd-trace-go/blob/v1/internal/container.go#L52
*

Reply:
Great information!!  Thank you.  Unfortunately, now I'm even more confused.

I think there's something significant here about the structure and expectations around `/proc/self/cgroup`.  Formally speaking, a process can participate in a different cgroup for each resource type, and it doesn't really offend my sensibilities to imagine that a container orchestration system would implement resource _pools_ instead of soft/hard caps.  In fact, this is a concept we had in HPC schedulers for 20+ years.  On the other hand, if a given container management system actually wanted to do this, I imagine it would preserve the integrity of the "container ID" concept by creating a sub-cgroup named with an appropriate ID for any given resource pool...  I don't know, I can speculate forever, but all I'm writing here is fictional until proven otherwise.

I guess what I'm saying is, it seems like this mechanism works in a fairly reliable way.  I'm not aware of any major complaints from the other tracer/profiler teams.  The _fact_ that it works is sort of mysterious to me.

Anyway, I don't think there's anything to be done about any of this, I'm just wasting your time and mine with pointless discussion.  Thanks again for sharing those sources, it was very informative for me (although I don't feel any closer to understanding the situation).  8)